packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -186,9 +187,12 @@ library LibProving {
                 }
             }
 
+            // Formal check of the instance equality. Shifted with
+            // 2 bytes because of the verifierId is also part of
+            // the evidence.proofs on a uint16 (2 bytes)
             if (
                 !LibBytesUtils.equal(
-                    LibBytesUtils.slice(evidence.proof, 0, 32),
+                    LibBytesUtils.slice(evidence.proofs, 2, 32),
```

Comment:
Can we also make `instance` one input variable to `verifyProofs`, then this logic can also be moved to the Verifier. `verifierId` should not be known here at all.